src/libraries/System.Numerics.Vectors/tests/GenericVectorTests.cs

```diff
@@ -846,6 +846,54 @@ public void VectorSingleEqualsNaNTest()
             var nan = new Vector<float>(float.NaN);
             Assert.True(nan.Equals(nan));
         }
+
+        [Fact]
+        public void VectorDoubleEqualsNonCanonicalNaNTest()
+        {
+            var maxSignificand = (1UL << 53) - 1;
+            // NaN with mantissa bits set
+            var snan = BitConverter.UInt64BitsToDouble(BitConverter.DoubleToUInt64Bits(double.NaN) | maxSignificand);
+            var nans = new double[]
+            {
+                double.CopySign(double.NaN, -0.0), // -qnan
+                double.CopySign(double.NaN, +0.0), // +qnan
+                double.CopySign(snan, -0.0),       // -snan
+                double.CopySign(snan, +0.0),       // +snan
```

Comment:
```suggestion
                double.NaN,                        // default nan
                double.CopySign(double.NaN, -0.0), // -qnan
                double.CopySign(double.NaN, +0.0), // +qnan
                double.CopySign(snan, -0.0),       // -snan
                double.CopySign(snan, +0.0),       // +snan
```
Maybe check default NaN too?

Reply:
the default is -qnan